org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/LinkedResourcesUpdater.java

```diff
@@ -39,30 +47,30 @@
  * Note that currently, we only include linked resources that are folders.
  */
 final class LinkedResourcesUpdater {
+    private static final QualifiedName LINKED_RESOURCES_KEY = new QualifiedName(CorePlugin.PLUGIN_ID, "linkedResources");
```

Comment:
Yeah I think that's a good idea and we might actually store more things there, like which folders are subprojects and which folder is the build folder. Then it doesn't matter anymore whether the build folder is actually there at project synchronization time.